extensions/3DTILES_binary_buffers/README.md

```diff
@@ -0,0 +1,166 @@
+# 3DTILES_binary_buffers
+
+## Overview
+
+This extension to 3D Tiles enables storage of binary data in external buffers.
+
+## Properties Reference
+
+---------------------------------------
+### 3DTILES_binary_buffers.bufferViews
+
+`bufferViews` provide a typed view into a `buffer`.
+
+**Properties**
+
+|   |Type|Description|Required|
+|---|----|-----------|--------|
+|**buffer**|`integer`|The index of the buffer.|☑️ Yes|
+|**byteOffset**|`integer`|The offset relative to the start of the buffer in bytes.|☑️ Yes|
+|**byteLength**|`integer`|The length of the bufferView in bytes.| ☑️ Yes|
+|**elementCount**|`integer`|The number of elements in the buffer view.| ☑️ Yes|
+|**elementByteOffsetsBufferView**|`integer`|The index of the bufferView containing byte offsets for each element. Must be defined for the STRING element type.|No|
+|**elementType**|`string`|Specifies if the attribute is a scalar, vector, matrix or string.|No, default is `SCALAR`.|
+|**componentType**|`string`|The datatype of components in the attribute. Must be defined for every other element type, except STRING (in which case it will be ignored).|No|
+
+
+#### Element Types
+
+| Element Type | No. of components |
+|:------------:|:-----------------:|
+| SCALAR | 1 |
+| STRING | 1 |
+| VEC2 | 2 |
+| VEC3 | 3 | 
+| VEC4 | 4 |
+| MAT2 | 4 |
+| MAT3 | 9 |
+| MAT4 | 16 |
+
+*Note: The `STRING` element type only stores UTF-8 encoded strings.*
+
+#### Component Types
+
+| Component | Size (bits) |
+|:---------:|:------------:|
+| BIT | 1 |
+| BYTE | 8 |
+| UNSIGNED_BYTE | 8 |
+| SHORT | 16 |
+| UNSIGNED_SHORT | 16 |
+| HALF_FLOAT | 16 |
+| INT | 32 |
+| UNSIGNED_INT | 32 |
+| FLOAT | 32 |
+| DOUBLE | 64 |
+
+
+*Note: The buffer views for the `BIT` component type must be padded with trailing `0`s to meet the nearest byte boundary.*
```

Comment:
The padding requirements should be defined for the other component types. In general properties must start on a byte offset that's a multiple of the size of the component type.

See https://github.com/CesiumGS/3d-tiles/tree/master/specification/TileFormats/BatchTable#padding

Reply:
What's the reason behind this? As long as the byte boundary is respected, the data can be read using byte offsets and byte lengths. What is gained by having the offset be a multiple of the size of the component type?

Reply:
Talked offline about this, but it's for performance reasons. In JavaScript typed arrays need to be aligned to their component type within the underlying array buffer which means unaligned data needs to be cloned into a new typed array. Proper alignment also provides performance benefits in systems level languages.

Early on glTF had the same rationale: https://github.com/KhronosGroup/glTF/issues/167

https://github.com/CesiumGS/cesium/blob/2a9ca28dcbb38eee7013505987516446a8eacb4f/Source/Scene/Batched3DModel3DTileContent.js#L370-L382 - this is where CesiumJS clones the typed array if it isn't aligned. 3D Tiles didn't always have byte alignment requirements which lead to slow paths like this.